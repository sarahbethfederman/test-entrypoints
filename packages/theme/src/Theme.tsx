import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
