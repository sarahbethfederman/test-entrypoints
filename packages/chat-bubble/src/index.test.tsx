import * as React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Theme from '@lendi-ui/theme';
import ChatBubble, { ChatBubbleProps } from './index';

const TestChatBubble = ({ direction }: ChatBubbleProps) => (
  <Theme>
    <ChatBubble direction={direction}>
      <ChatBubble.Content className="cutomer-class" itemID="chat-content">
        Can you help me to get a loan?
      </ChatBubble.Content>
      <ChatBubble.Footer>12:34PM Read</ChatBubble.Footer>
    </ChatBubble>
  </Theme>
);

describe('ChatBubble', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render ChatBubble component', () => {
    const { getByTestId } = render(<TestChatBubble direction="left" />);

    expect(getByTestId('chat-bubble'));
    expect(getByTestId('chat-bubble')).toMatchSnapshot();
  });

  it('should render ChatBubble component', () => {
    const { getByTestId } = render(<TestChatBubble direction="left" />);

    expect(getByTestId('chat-bubble-content'));
    expect(getByTestId('chat-bubble-content').className).toContain('cutomer-class');
  });

  it('should render ChatBubble component', () => {
    const { getByTestId } = render(<TestChatBubble direction="left" />);

    expect(getByTestId('chat-bubble-content'));
    expect(getByTestId('chat-bubble-content').getAttribute('itemID')).toEqual('chat-content');
  });
});
