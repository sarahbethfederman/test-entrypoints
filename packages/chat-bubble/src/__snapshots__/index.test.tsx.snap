// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBubble should render ChatBubble component 1`] = `
.c0 {
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c1 {
  position: relative;
  border-radius: 0 5px 5px 5px;
}

.c1:before {
  content: '';
  width: 20px;
  height: 20px;
  top: 0px;
  right: 100%;
  position: absolute;
  background: inherit;
}

.c1:after {
  content: '';
  width: 20px;
  height: 40px;
  border-radius: 0 20px 20px 0;
  top: 0px;
  right: 100%;
  position: absolute;
  background-color: rgba(255,255,255,1);
}

<div
  class="c0"
  data-testid="chat-bubble"
  direction="left"
>
  <div
    class="c1 cutomer-class"
    data-testid="chat-bubble-content"
    direction="left"
    itemid="chat-content"
  >
    Can you help me to get a loan?
  </div>
  <span>
    12:34PM Read
  </span>
</div>
`;
